### PR TITLE
delete plugin rootfs on `docker plugin rm`

### PR DESCRIPTION
**Pull request by vieux**
ping @anusha-ragunathan @tiborvass

### CODE REVIEW

**Comment by tiborvass** [comment]
LOLGTM

**Comment by cpuguy83** [comment]
```
01:37:46 FAIL: docker_cli_daemon_experimental_test.go:83: DockerDaemonSuite.TestDaemonKillLiveRestoreWithPlugins
01:37:46 
01:37:46 [d25b86087d036] waiting for daemon to start
01:37:46 [d25b86087d036] daemon started
01:37:46 docker_cli_daemon_experimental_test.go:90:
01:37:46     c.Fatalf("Could not install plugin: %v %s", err, out)
01:37:46 ... Error: Could not install plugin: exit status 1 Error response from daemon: lstat /go/src/github.com/docker/docker/bundles/1.13.0-dev/test-integration-cli/d25b86087d036/root/plugins/ab9e637da9af9709f463263bac93054a2bded0888aa54f0f4f196ed854ecec11/rootfs/run/docker/plugins: no such file or directory
01:37:46 
```

**Comment by vieux** [comment]
@cpuguy83 the error was unrelated I believe, the plugin we use for test was misbehaving at the time, I restarted and it passes now

**Comment by vieux** [comment]
@anusha-ragunathan I see a `p.RemoveFromDisk` in manager.go, but it's only removing runRoot when the container state change, I believe we need to remove libRoot/id as well

File: plugin/backend.go [hunk 1]
```diff
@@ -162,6 +162,7 @@ func (pm *Manager) Remove(name string, config *types.PluginRmConfig) error {
 	}
 
 	pm.pluginStore.Remove(p)
+	os.RemoveAll(filepath.Join(pm.libRoot, p.PluginObj.ID))
```
Comment (anusha-ragunathan) [inline plugin/backend.go]: Lets try and avoid direct access to the PluginObj from manager and use a wrapper instead. Eventually, we should convert all Plugin struct fields to internal types.

Comment (vieux) [inline plugin/backend.go]: right, it's pushed.

**Comment by anusha-ragunathan** [comment]
While you are here, can you also fix the test at https://github.com/docker/docker/blob/master/integration-cli/docker_cli_plugins_test.go#L43 ?

**Comment by vieux** [comment]
@anusha-ragunathan done

**Comment by vieux** [comment]
thanks @anusha-ragunathan for tracking down the bad test

**Comment by anusha-ragunathan** [comment]
LGTM (granted all tests pass)

**Pull request merged**